internal/core/ports/operation_lease_storage.go

```diff
@@ -0,0 +1,43 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package ports
+
+import (
+	"context"
+	"time"
+
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+)
+
+type OperationLeaseStorage interface {
+	// Grant takes a lease from an operation. NOTE: only one worker can have the operation lease, meaning that if a GrantLease called for an operation that already has it, it will return an error.
```

Comment:
@caiordjesus This comment over here should start with the function name (GrantLease) this is a convention for commentaries in go, you can find more info [here](https://go.dev/doc/effective_go), in `Commentary` section